crossbeam-channel/tests/golang.rs

```diff
@@ -1507,5 +1507,59 @@ mod chan {
 
 // https://github.com/golang/go/blob/master/test/ken/chan1.go
 mod chan1 {
-    // TODO
+    use super::*;
+
+    // sent messages
+    const N: usize = 1000;
+    // receiving "goroutines"
+    const M: usize = 10;
+    // channel buffering
+    const W: usize = 2;
+
+    fn r(c: Chan<usize>, m: usize, h: Arc<Mutex<[usize; N]>>) {
+        loop {
+            select! {
+                recv(c.rx()) -> rr => {
+                    if let Ok(r) = rr {
+                        let mut data = h.lock().unwrap();
+                        if data[r] != 1 {
+                            println!("r\nm={}\nr={}\nh={}\n", m, r, data[r]);
+                            panic!("fail")
+                        }
+                        data[r] = 2;
+                        drop(data);
```

Comment:
Someone who is more experienced with Rust than I am probably knows why, but this blocks unless I manually drop.
Any thoughts?

Reply:
There may be an odd interaction with select macro. I'll take a look.

Reply:
@taiki-e What do I need to do to get this PR merged in?

Reply:
Even if I remove `drop(data)` in the `r` function, I couldn't reproduce block. I can reproduce block by removing the `drop(data)` in the `s` function, but it will not block if I disable parallel tests (`cargo test -- --test-threads=1`) or run the test individually (`cargo test --test golang -- chan1::main`).

@SkamDart Could you remove `drop(data)` in the `r` function and leave the link to this comment on the `drop(data)` in the `s` function as a comment? (Otherwise, LGTM)